tests/Tests/200__monitor_and_manage/200__metrics/201__billing_metrics.robot

```diff
@@ -17,121 +20,253 @@ ${METRIC_RHODS_UNDEFINED}           cluster:usage:consumption:rhods:undefined:se
-  ${authorization_required} =  Is Service Account Authorization Required
-  Run Keyword If  ${authorization_required}  Authorize jupyterhub service account
-  Common.End Web Test
+    Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+    Login To RHODS Dashboard    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Wait for RHODS Dashboard to Load
+    ${version-check} =    Is RHODS Version Greater Or Equal Than    1.4.0
+    IF    ${version-check}==True
+        Launch JupyterHub From RHODS Dashboard Link
+    ELSE
+        Launch JupyterHub From RHODS Dashboard Dropdown
+    END
+    Login To Jupyterhub    ${TEST_USER.USERNAME}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+    Page Should Not Contain    403 : Forbidden
+    ${authorization_required} =    Is Service Account Authorization Required
+    Run Keyword If    ${authorization_required}    Authorize jupyterhub service account
+    Common.End Web Test
+
+Log In N Users To JupyterLab And Launch A Notebook For Each Of Them
+    [Documentation]    Log in N users and run notebook for each of them
+    [Arguments]    ${list_of_usernames}
+    FOR    ${n}    IN    @{list_of_usernames}
+        Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+        Login To RHODS Dashboard    ${n}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+        Wait for RHODS Dashboard to Load
+        Launch JupyterHub From RHODS Dashboard Link
+        Login To Jupyterhub    ${n}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+        Page Should Not Contain    403 : Forbidden
+        ${authorization_required} =    Is Service Account Authorization Required
+        Run Keyword If    ${authorization_required}    Authorize jupyterhub service account
+        Fix Spawner Status
+        Spawn Notebook With Arguments
+    END
+    [Teardown]    SeleniumLibrary.Close All Browsers
+
+CleanUp JupyterHub For N users
+    [Documentation]    Cleans JupyterHub for N users
+    [Arguments]    ${list_of_usernames}
+    SeleniumLibrary.Close All Browsers
+    FOR    ${n}    IN    @{list_of_usernames}
+        Open Browser    ${ODH_DASHBOARD_URL}    browser=${BROWSER.NAME}    options=${BROWSER.OPTIONS}
+        Login To RHODS Dashboard    ${n}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+        Wait for RHODS Dashboard to Load
+        ${version-check} =    Is RHODS Version Greater Or Equal Than    1.4.0
+        IF    ${version-check}==True
+            Launch JupyterHub From RHODS Dashboard Link
+        ELSE
+            Launch JupyterHub From RHODS Dashboard Dropdown
+        END
+        Login To Jupyterhub    ${n}    ${TEST_USER.PASSWORD}    ${TEST_USER.AUTH_TYPE}
+        Page Should Not Contain    403 : Forbidden
+        ${authorization_required} =    Is Service Account Authorization Required
+        Run Keyword If    ${authorization_required}    Authorize jupyterhub service account
```

Comment:
You might need to call `Clean Up Server` to clean all files in the notebook

Reply:
I don't think so because we are just Spawn the notebook not running any script or jupyter notebook
But if you want we can add that one